tests/functional/flows/test_create_flow.py

```diff
@@ -0,0 +1,220 @@
+        "name": "Pete Wrigley",
+        "identity_provider": "c8abac57-560c-46c8-b386-f116ed8793d5",
+        "organization": "KrebStar Corp",
+        "status": "used",
+        "email": "pete@kreb.star",
+    },
+    "nona": {
+        "username": "nona@wellsville.gov",
+        "name": "Nona F. Mecklenberg",
+        "identity_provider": "c8abac57-560c-46c8-b386-f116ed8793d5",
+        "organization": "The City of Wellsville",
+        "status": "used",
+        "email": "nona@wellsville.gov",
+    },
+    "artie": {
+        "username": "artie@super.hero",
+        "name": "The Strongest Man in the World",
+        "identity_provider": "c8abac57-560c-46c8-b386-f116ed8793d5",
+        "organization": "Personal Superheroes",
+        "status": "used",
+        "email": "artie@super.hero",
+    },
+    "monica": {
+        "username": "monica@kreb.scouts",
+        "name": "Monica Perling",
+        "identity_provider": "c8abac57-560c-46c8-b386-f116ed8793d5",
+        "organization": "Kreb Scouts",
+        "status": "used",
+        "email": "monica@kreb.scouts",
+    },
+}
+
+SPECIAL_PRINCIPALS = ["public", "all_authenticated_users"]
+
+
+def get_identity(id, name):
+    """
+    Return an identity dict using the provided id for the user corresponding
+    to the provided name.
+    """
+    identity = FLOW_IDENTITIES[name].copy()
+    identity["id"] = id
+    return identity
+
+
+def assign_identities_for_principal_set(principal_set, existing_identities):
+    """
+    Return a list of identities for the provided principal set, updating the provided
+    existing_identities dict with any new identities that are created.
+
+    principal_set is a list of principals in the request
+    existing_identities is a dict mapping principal URNs to identities
+    """
+    identity_set = []
+    # Randomize the names for each principal set
+    available_identities = list(FLOW_IDENTITIES.keys())
+    shuffle(available_identities)
+
+    # Iterate over the principals in the request
+    for index, principal in enumerate(principal_set):
+        if principal in existing_identities:
+            # Use the existing identity if it's already been assigned
+            identity_set.append(existing_identities[principal])
+            continue
+
+        if principal not in SPECIAL_PRINCIPALS:
+            # Attempt to assign distinct identities to each principal
+            identity = get_identity(
+                principal.split(":")[-1],
+                available_identities[index % len(available_identities)],
+            )
+            identity_set.append(identity)
+            existing_identities[principal] = identity
+
+    return identity_set
+
+
+def value_for_field_from_output(name, output):
+    """
+    Return the value for a specified field from the output of a command.
+    """
+    match = re.search(rf"^{name}:[^\S\n\r]+(?P<value>.*)$", output, flags=re.M)
+    assert match is not None
+    return match.group("value")
```

Comment:
I like this helper and I'm thinking about how we can make this into a fixture which we can expose more globally within the testsuite. I think I did something like this at some point in the distant past, but not with a good interface. So this is a problem I know we keep seeing and solving per-test, and I want to put it into our hands more consistently.